pandas/core/algorithms.py

```diff
@@ -105,6 +105,9 @@ def _ensure_data(values, dtype=None):
         else:
             # Datetime
             from pandas import DatetimeIndex
+            from pandas.core.arrays import unwrap_reshapeable
```

Comment:
Is this for something like `factorize(EA)`? Shouldn't the EA (or your wrapper) do this in `_values_for_factorize`?

Reply:
values_for_factorize might do it, ill check.  But at this point we dont _necessarily_ have a EA, so need a conditional un-wrapper.

Making DTA validate that inputs are 1D can be done separately from the rest of this, which should resolve this particular part of the diff